paf-mvp-frontend/src/lib/paf-lib.ts

```diff
@@ -238,8 +248,11 @@ export const refreshIdsAndPreferences = async ({
 
       return {
         status: pafStatus,
-        data: fromClientCookieValues(strIds, strPreferences),
+        data: currentPafData,
       };
+    } else if (strIds || strPreferences) {
+      removeCookie(Cookies.preferences);
```

Comment:
Is it really necessary?
I understand you already stored the current value when building `triggerNotification` function, so you don't need the cookies anymore.

But I see a couple of issues with removing the cookies here:
1. if you trigger a redirect line 326, you will have no more cookies to compare to, after the redirect. So I don't see how you can show a notification after a redirect 🤔 
2. if anything fails when doing a REST call (lines 27x), then you will have lost previous data.

So, unless I'm wrong, I think it is a mistake to remove the cookies here.

Reply:
Indeed